testutils/src/main/scala/za/co/absa/enceladus/testutils/e2eSparkRunner/E2ESparkRunner.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2018-2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.testutils.e2eSparkRunner
+
+import org.apache.log4j.{LogManager, Logger}
+import za.co.absa.enceladus.testutils.HelperFunctions
+import za.co.absa.enceladus.testutils.models.ProcessMeasurement
+
+import sys.process._
+import scala.io.Source
+
+object E2ESparkRunner {
+  private val log: Logger = LogManager.getLogger(this.getClass)
+
+  private val stdJarPath: String = "$DCE_JAR_PATH/$DCE_STD_SPT_JAR"
+  private val confJarPath: String = "$DCE_JAR_PATH/$DCE_CNFRM_SPT_JAR"
+  private val testUtilsJarPath: String = "$DCE_JAR_PATH/$TEST_UTILS_JAR"
+
+  private val stdClass = "--class za.co.absa.enceladus.standardization.StandardizationJob"
+  private val confClass = "--class za.co.absa.enceladus.conformance.DynamicConformanceJob"
+  private val compClass = "--class za.co.absa.enceladus.testutils.datasetComparison.ComparisonJob"
+
+  private def runBashCmd(bashCmd: String): String = {
+    (s"echo $bashCmd" #| "bash").!!
```

Comment:
Wow. What does this do? Runs an arbitrary command in OS?

Reply:
Yes, it will run our spark jobs like we normally would on bash. I got it from a Scala cookbook by [Alvin Alexander](https://alvinalexander.com/scala/scala-execute-exec-external-system-commands-in-scala).

Sending the command through the pipe to bash was me solving the problem of bash "swallowing up" some special characters and then running the command weirdly.

Reply:
I had to ask as well 😉